### ISSUE TITLE
Permission Error after moving to a networkdrive

### ISSUE THREAD

**Issue by dav354** [meeb/tubesync#240]
Hey, today I changed the layout of my Server so now the location where the videos are stored is now a NFS share where I've copied the existing files (the /downloads location). Now when I try to access the UI of tubesync it shows me a permission error. I did not touch any permissions... but I dont know what to do

![2022-05-17_22-23](https://user-images.githubusercontent.com/98947769/168904247-4580599a-8c43-4e9e-98b9-96270ae7291c.png)

```
Environment:


Request Method: GET
Request URL: http://localhost/

Django Version: 3.2.13
Python Version: 3.9.2
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'sass_processor',
 'background_task',
 'common',
 'sync']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware',
 'common.middleware.MaterializeDefaultFieldsMiddleware',
 'common.middleware.BasicAuthMiddleware']


Template error:
In template /app/common/templates/base.html, error at line 11
   13
   1 : {% load static %}{% load sass_tags %}<!DOCTYPE html>
   2 : <html>
   3 :   <head>
   4 :     <meta charset="utf-8" />
   5 :     <style>
   6 :       html{
   7 :         visibility: hidden;
   8 :         opacity: 0;
   9 :       }
   10 :     </style>
   11 :     <link type="text/css" rel="stylesheet" href=" {% sass_src 'styles/tubesync.scss' %} " media="screen,projection"/>
   12 :     <link rel="icon" href="{% static 'images/favicon.ico' %}" sizes="21x21" type="image/x-icon">
   13 :     <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
   14 :     <title>TubeSync - {% block headtitle %}Synchronize YouTube to your local media server{% endblock %}</title>
   15 :   </head>
   16 : 
   17 :   <body>
   18 : 
   19 :     <div class="app">
   20 : 
   21 :       <header>


Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.9/dist-packages/django/core/handlers/base.py", line 204, in _get_response
    response = response.render()
  File "/usr/local/lib/python3.9/dist-packages/django/template/response.py", line 105, in render
    self.content = self.rendered_content
  File "/usr/local/lib/python3.9/dist-packages/django/template/response.py", line 83, in rendered_content
    return template.render(context, self._request)
  File "/usr/local/lib/python3.9/dist-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "/usr/local/lib/python3.9/dist-packages/django/template/base.py", line 170, in render
    return self._render(context)
  File "/usr/local/lib/python3.9/dist-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.9/dist-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.9/dist-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.9/dist-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/usr/local/lib/python3.9/dist-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.9/dist-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.9/dist-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.9/dist-packages/sass_processor/templatetags/sass_tags.py", line 35, in render
    path = self.sass_processor(self.sass_processor.resolve_path(context))
  File "/usr/local/lib/python3.9/dist-packages/sass_processor/processor.py", line 113, in __call__
    self.source_storage.delete(css_filename)
  File "/usr/local/lib/python3.9/dist-packages/django/core/files/storage.py", line 318, in delete
    os.remove(name)

Exception Type: PermissionError at /
Exception Value: [Errno 13] Permission denied: '/app/static/styles/tubesync.css'
```

**Comment by meeb** [comment]
That error means the container can't write to the local container file system, the `tubesync.css` file is dynamically built and written to the local disk. This can only be a permissions issue with where you're running the container. Try stopping and removing the container, then checking the permissions of the drive where containers are run, and starting TubeSync again.

This isn't really a TubeSync issue, this is an issue with your server. I would suspect that other containers that write to the local disk in the container are now also failing if you have any.

**Comment by dav354** [comment]
That is what I've had in mind, but tubesync is the only container having this issue. I have around 20 containers running and all others are fine. I tried the last days all kind of different permissions, but it dont*t work. It
s so bad that I can't figure it out, because I really like tubesync...

**Comment by meeb** [comment]
Your issue can only be a local permissions issue, sorry there isn't much else I can say. Did you try actually fully deleting the container and recreating it? You won't lose any data providing the SQLite database in the `config` volume isn't deleted. If that still doesn't work, you can try dropping a shell into the container and resetting permissions manually. This:

```bash
$ docker exec -ti tubesync bash
root@aabbccddeeffgg:/app# chown -R app:app /app/common/static/styles/
root@aabbccddeeffgg:/app# chown -R app:app /app/static/styles/
root@aabbccddeeffgg:/app# rm /app/static/styles/tubesync.css
root@aabbccddeeffgg:/app# rm /app/common/static/styles/tubesync.css
```

for example should trigger `tubesync.css` to be regenerated the next time you load the dashboard.

Also note TubeSync (and other containers) will likely have issues if you store either the docker root volumes or the `config` directory on a network share, SQLite doesn't work very well over network shares. If you must use a network share for the `config` directory I would suggest you migrate the database to use Postgres or MySQL. There's guides linked in the readme on how to do this.

**Comment by dav354** [comment]
Yes, I tried reinstalling the container but it changes nothing. This weekend I'll try to set the permissions manually as you mentioned. And to clarify I only moved the download path, not the config path and yes, I already use a Mariadb database. And that's the thing I think is weird. The css file shouldn't be related to the download path right? It should be in the config path and doesn't relate to changing the downloads path. If I change the path of the downloads back to the disk where the config lives everything is alright
